src/stan/services/optimize/laplace_sample.hpp

```diff
@@ -70,54 +70,53 @@ void laplace_sample(const Model& model, const Eigen::VectorXd& theta_hat,
   math::internal::finite_diff_hessian_auto(log_density_fun, theta_hat, log_p,
                                            grad, hessian);
   if (refresh > 0) {
-    logger.info(log_density_msgs);
-    logger.info("\n");
+    if (log_density_msgs.peek() != std::char_traits<char>::eof())
+      logger.info(log_density_msgs);
   }
 
   // calculate Cholesky factor and inverse
   interrupt();
   if (refresh > 0) {
     logger.info("Calculating inverse of Cholesky factor");
-    logger.info("\n");
   }
   Eigen::MatrixXd L_neg_hessian = (-hessian).llt().matrixL();
   interrupt();
   Eigen::MatrixXd inv_sqrt_neg_hessian = L_neg_hessian.inverse().transpose();
   interrupt();
   Eigen::MatrixXd half_hessian = 0.5 * hessian;
 
-  // generate draws and output to sample writer
   if (refresh > 0) {
     logger.info("Generating draws");
-    logger.info("\n");
   }
+  // generate draws
+  std::stringstream refresh_msg;
   boost::ecuyer1988 rng = util::create_rng(random_seed, 0);
   Eigen::VectorXd draw_vec;  // declare draw_vec, msgs here to avoid re-alloc
   for (int m = 0; m < draws; ++m) {
     interrupt();  // allow interpution each iteration
     if (refresh > 0 && m % refresh == 0) {
-      logger.info("iteration: ");
-      logger.info(std::to_string(m));
-      logger.info("\n");
+      refresh_msg << "iteration: " << std::to_string(m);
```

Comment:
The `refresh_msg` variable is created outside the loop.  That means it needs to get cleared somehow so that it's not reused and the message doesn't just keep getting longer.  I don't see that being done in the diff.   I'd suggest moving the declaration inside the conditional to keep things easy to follow.